// Generated by Haxe 4.2.2

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nMorph.framework.common.exceptions {
	public class ExceptionHelper : global::haxe.lang.HxObject {
		
		public ExceptionHelper(global::haxe.lang.EmptyObject empty) {
		}
		
		
		public ExceptionHelper() {
			global::nMorph.framework.common.exceptions.ExceptionHelper.__hx_ctor_nMorph_framework_common_exceptions_ExceptionHelper(this);
		}
		
		
		protected static void __hx_ctor_nMorph_framework_common_exceptions_ExceptionHelper(global::nMorph.framework.common.exceptions.ExceptionHelper __hx_this) {
		}
		
		
		public static void LogicException(int Code, string Msg) {
			try {
				throw new global::nMorph.framework.common.exceptions.LogicException(((int) (Code) ), ((string) (Msg) ));
			}
			catch (global::System.Exception _g){
				{
					global::haxe.NativeStackTrace.exception = ((global::System.Exception) (((object) (_g) )) );
				}
				
				object e = ((object) (global::haxe.Exception.caught(_g).unwrap()) );
				throw ((global::System.Exception) (global::haxe.Exception.thrown(e)) );
			}
			
			
		}
		
		
		public static void SecurityException(int Code, string Msg) {
			try {
				throw new global::nMorph.framework.common.exceptions.SecurityException(((int) (Code) ), ((string) (Msg) ));
			}
			catch (global::System.Exception _g){
				{
					global::haxe.NativeStackTrace.exception = ((global::System.Exception) (((object) (_g) )) );
				}
				
				object e = ((object) (global::haxe.Exception.caught(_g).unwrap()) );
				throw ((global::System.Exception) (global::haxe.Exception.thrown(e)) );
			}
			
			
		}
		
		
	}
}


